tests/remote_resources/xtcp_ping_pong.py

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/python
+import socket
+import random
+import string
+import time
+import argparse
+import sys
+from multiprocessing import Pool
+
+def print_errors(failure_list):
+    for process_failures in failure_list:
+        if len(process_failures) > 0:
+            print ''.join(process_failures)
+
+    print "Lost_packets: {}".format(sum(map(len, failure_list)))
+
+def format_message(message, width):
+    return '\n  '.join([message[i:i+width] for i in range(0, len(message), width)])
+
+# Limit reconnect attempts
+def attempt_connect(failures, process_port, sock=None):
+    for _ in range(3):
+        try:
+            sock = socket.socket(socket.AF_INET, args.protocol)
+            sock.settimeout(5) # seconds
+            sock.connect((args.ip, process_port))
+            return (failures, sock)
+
+        except socket.error as err:
+            failures.append('ERROR: Could not connect to remote port\n' +
+                            '  Local port:    {}\n'.format(sock.getsockname()[1]) +
+                            '  Remote port:   {}\n'.format(process_port) +
+                            '  Error message: {}\n'.format(err)
+                            )
+
+    return (failures, None)
+
+def process_test(process_port):
+    random.seed()
```

Comment:
It'd be helpful if the pseudo random elements of the testing were easily repeatable - if the test were to fail on the CI system but then pass when run locally debugging the issue will be quite a lot harder!

Ideally running the test multiple times would provide additional coverage, whilst still ensuring that if a specific build gets rerun on the CI system the same test cases are exercised...

You could seed the random modele (at the start of  with a repeatable but non fixed value, such as the current git hash of the repo, so that the test cases for each snapshot in the CI system are fixed (and reproducible locally).
To increase coverage we can always run more test cases, though some thought will need to be given as to where this done, given the use of a process pool in this test.

I've used the following to seed random before:
```python
# Use the current git hash to seed the random number generator,
# to avoid the number of test cases for a particular snapshot increasing
# each time the view is built in the CI system.
script_location = os.path.dirname(os.path.realpath(__file__))
stdout, stderr = xmostest.call_get_output(['git', 'rev-parse', 'HEAD'],
                                          cwd=script_location)
random.seed(int(stdout[0].strip(), 16)) # Git hash is hexadecimal string
```